src/analysis_and_optimization/Dataflow_utils.ml

```diff
@@ -174,7 +174,7 @@ let build_cf_graphs ?(flatten_loops = false) ?(blocks_after_body = true)
                   Set.Poly.diff substmt_state_unlooped.breaks in_state.breaks ]
           in
           ({substmt_state_unlooped with exits= loop_exits}, loop_predecessors)
-      | Block _ when blocks_after_body ->
+      | (Block _ | Profile _) when blocks_after_body ->
```

Comment:
This was identical to [line 140](https://github.com/stan-dev/stanc3/blob/da5c6175f36e04d17464aa4831ce82a1ae4d44b3/src/analysis_and_optimization/Dataflow_utils.ml#L140). Should that change too? (And I'm not sure why this doesn't have SList...)

Reply:
I don't believe SLists exist until _after_ optimization, which is why they're not there.

Good catch on line 140, I believe it should also be done there

Reply:
This also fixed the build error due to the SoA optimization.